File: js/922.sort-array-by-parity-ii.js
/*
 * @lc app=leetcode id=922 lang=javascript
 *
 * [922] Sort Array By Parity II
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParityII = function(nums) {
  let output = Array(nums.length)
  let iEven = 0
  let iOdd = 1

  for (let i=0;i<nums.length;i++) {
    if (nums[i] % 2 === 0) {
      output[iEven] = nums[i]
      iEven += 2
    } else {
      output[iOdd] = nums[i]
      iOdd += 2
    }
  }

  return output
};
// @lc code=end

File: 922.sort-array-by-parity-ii.py
#
# @lc app=leetcode id=922 lang=python3
#
# [922] Sort Array By Parity II
#

# @lc code=start
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        result = {}
        iEven = 0
        iOdd = 1
        for num in nums:
            if num%2==1:
                result[iOdd] = num
                iOdd += 2
            else:
                result[iEven] = num
                iEven += 2
        return [result[i] for i in sorted(result.keys())]
# @lc code=end

